### PR TITLE
[api] get account modules

### PR DESCRIPTION
**Pull request by xli**
#9193 get account modules API.

Endpoint URL path: `/accounts/<address>/modules`
Returns a list of MoveModule JSON object.

The MoveModule object includes module exposed functions (including friend functions), structs and and friend modules.

Example module JSON:

```JavaScript
{
    "address": "0x1",
    "name": "DiemConfig",
    "friends": [
        {
            "address": "0x1",
            "name": "DiemConsensusConfig"
        },
        {
            "address": "0x1",
            "name": "DiemSystem"
        },
        {
            "address": "0x1",
            "name": "DiemTransactionPublishingOption"
        },
        {
            "address": "0x1",
            "name": "DiemVMConfig"
        },
        {
            "address": "0x1",
            "name": "DiemVersion"
        },
        {
            "address": "0x1",
            "name": "RegisteredCurrencies"
        }
    ],
    "exposed_functions": [
        {
            "name": "get",
            "visibility": "public",
            "generic_type_params": [
                {
                    "constraints": [
                        "copy",
                        "drop",
                        "store"
                    ]
                }
            ],
            "params": [],
            "return": [
                {
                    "type": "generic_type_param",
                    "index": 0
                }
            ]
        },
        {
            "name": "initialize",
            "visibility": "public",
            "generic_type_params": [],
            "params": [
                {
                    "type": "reference",
                    "mutable": false,
                    "to": {
                        "type": "signer"
                    }
                }
            ],
            "return": []
        },
        {
            "name": "publish_new_config",
            "visibility": "friend",
            "generic_type_params": [
                {
                    "constraints": [
                        "copy",
                        "drop",
                        "store"
                    ]
                }
            ],
            "params": [
                {
                    "type": "reference",
                    "mutable": false,
                    "to": {
                        "type": "signer"
                    }
                },
                {
                    "type": "generic_type_param",
                    "index": 0
                }
            ],
            "return": []
        },
        {
            "name": "publish_new_config_and_get_capability",
            "visibility": "friend",
            "generic_type_params": [
                {
                    "constraints": [
                        "copy",
                        "drop",
                        "store"
                    ]
                }
            ],
            "params": [
                {
                    "type": "reference",
                    "mutable": false,
                    "to": {
                        "type": "signer"
                    }
                },
                {
                    "type": "generic_type_param",
                    "index": 0
                }
            ],
            "return": [
                {
                    "type": "struct",
                    "address": "0x1",
                    "module": "DiemConfig",
                    "name": "ModifyConfigCapability",
                    "generic_type_params": [
                        {
                            "type": "generic_type_param",
                            "index": 0
                        }
                    ]
                }
            ]
        },
        {
            "name": "reconfigure",
            "visibility": "public",
            "generic_type_params": [],
            "params": [
                {
                    "type": "reference",
                    "mutable": false,
                    "to": {
                        "type": "signer"
                    }
                }
            ],
            "return": []
        },
        {
            "name": "set",
            "visibility": "friend",
            "generic_type_params": [
                {
                    "constraints": [
                        "copy",
                        "drop",
                        "store"
                    ]
                }
            ],
            "params": [
                {
                    "type": "reference",
                    "mutable": false,
                    "to": {
                        "type": "signer"
                    }
                },
                {
                    "type": "generic_type_param",
                    "index": 0
                }
            ],
            "return": []
        },
        {
            "name": "set_with_capability_and_reconfigure",
            "visibility": "friend",
            "generic_type_params": [
                {
                    "constraints": [
                        "copy",
                        "drop",
                        "store"
                    ]
                }
            ],
            "params": [
                {
                    "type": "reference",
                    "mutable": false,
                    "to": {
                        "type": "struct",
                        "address": "0x1",
                        "module": "DiemConfig",
                        "name": "ModifyConfigCapability",
                        "generic_type_params": [
                            {
                                "type": "generic_type_param",
                                "index": 0
                            }
                        ]
                    }
                },
                {
                    "type": "generic_type_param",
                    "index": 0
                }
            ],
            "return": []
        }
    ],
    "structs": [
        {
            "name": "Configuration",
            "is_native": false,
            "abilities": [
                "key"
            ],
            "generic_type_params": [],
            "fields": [
                {
                    "name": "epoch",
                    "type": {
                        "type": "u64"
                    }
                },
                {
                    "name": "last_reconfiguration_time",
                    "type": {
                        "type": "u64"
                    }
                },
                {
                    "name": "events",
                    "type": {
                        "type": "struct",
                        "address": "0x1",
                        "module": "Event",
                        "name": "EventHandle",
                        "generic_type_params": [
                            {
                                "type": "struct",
                                "address": "0x1",
                                "module": "DiemConfig",
                                "name": "NewEpochEvent",
                                "generic_type_params": []
                            }
                        ]
                    }
                }
            ]
        },
        {
            "name": "DiemConfig",
            "is_native": false,
            "abilities": [
                "store",
                "key"
            ],
            "generic_type_params": [
                {
                    "constraints": [
                        "copy",
                        "drop",
                        "store"
                    ],
                    "is_phantom": false
                }
            ],
            "fields": [
                {
                    "name": "payload",
                    "type": {
                        "type": "generic_type_param",
                        "index": 0
                    }
                }
            ]
        },
        {
            "name": "DisableReconfiguration",
            "is_native": false,
            "abilities": [
                "key"
            ],
            "generic_type_params": [],
            "fields": [
                {
                    "name": "dummy_field",
                    "type": {
                        "type": "bool"
                    }
                }
            ]
        },
        {
            "name": "ModifyConfigCapability",
            "is_native": false,
            "abilities": [
                "store",
                "key"
            ],
            "generic_type_params": [
                {
                    "constraints": [],
                    "is_phantom": true
                }
            ],
            "fields": [
                {
                    "name": "dummy_field",
                    "type": {
                        "type": "bool"
                    }
                }
            ]
        },
        {
            "name": "NewEpochEvent",
            "is_native": false,
            "abilities": [
                "drop",
                "store"
            ],
            "generic_type_params": [],
            "fields": [
                {
                    "name": "epoch",
                    "type": {
                        "type": "u64"
                    }
                }
            ]
        }
    ]
}
```

Example module has script functions
```JavaScript
{
    "address": "0x1",
    "name": "PaymentScripts",
    "friends": [],
    "exposed_functions": [
        {
            "name": "peer_to_peer_by_signers",
            "visibility": "script",
            "generic_type_params": [
                {
                    "constraints": []
                }
            ],
            "params": [
                {"type": "signer"},
                {"type": "signer"},
                {"type": "u64"},
                {
                    "type": "vector",
                    "items": {"type": "u8"}
                }
            ],
            "return": []
        },
        {
            "name": "peer_to_peer_with_metadata",
            "visibility": "script",
            "generic_type_params": [
                {
                    "constraints": []
                }
            ],
            "params": [
                {"type": "signer"},
                {"type": "address"},
                {"type": "u64"},
                {
                    "type": "vector",
                    "items": {"type": "u8"}
                },
                {
                    "type": "vector",
                    "items": {"type": "u8"}
                }
            ],
            "return": []
        }
    ],
    "structs": []
}
```

### CODE REVIEW

File: api/src/accounts.rs [hunk 1]
```diff
@@ -46,25 +64,36 @@ impl GetAccountResources {
...
+        self.context
+            .get_account_state(&self.address, self.ledger_info.version())
+    }
 }
 
 #[cfg(any(test))]
 mod tests {
-    use crate::test_utils::{new_test_context, send_request};
-    use serde_json::{json, Value};
+    use crate::test_utils::{assert_json, find_value, new_test_context, send_request};
```
Comment (dimroc) [inline api/src/accounts.rs:95]: nit: only bring in `test_utils` and on invocation of method, have the prefix to show to other devs where it's coming from and it's test intention.

ie:

```rust
use crate::test_utils;
...
let context = test_utils::new_context();
...
test_utils::assert_json(...)
...
let resp = test_utils::send_request(context, "GET", &account_resources(address), 200).await;
```

**Review by dimroc — APPROVED** [review]
💯

File: api/src/accounts.rs (hunk 1 above)
Comment (xli) [reply api/src/accounts.rs:95]: I think a good IDE solves this pretty well by providing source code jump functions.
The import is primary for shorting the code.
Let me think about this, may address in the following diff.

Comment (dimroc) [reply api/src/accounts.rs:95]: Sounds good. My opinion: jumping to the definition requires a second step per line of code that's of interest. The `test_utils::` prefix is seen before jumping for every line.

**Pull request merged**